02 - Web UI Template/CognitiveSearch.UI/Search/DocumentSearchClient.cs

```diff
@@ -337,67 +337,71 @@ public DocumentResult GetDocuments(string q, SearchFacet[] searchFacets, int cur
-            if (decodedPath.ToLower().Contains(containerAddresses[1])) { tokenToUse = tokens[1]; storageIndex = 1; }
-            else if (decodedPath.ToLower().Contains(containerAddresses[2])) { tokenToUse = tokens[2]; storageIndex = 2; }
-            else { tokenToUse = tokens[0]; storageIndex = 0; }
+            if (decodedPath.ToLower().Contains(s_containerAddresses[1])) { tokenToUse = s_tokens[1]; storageIndex = 1; }
+            else if (decodedPath.ToLower().Contains(s_containerAddresses[2])) { tokenToUse = s_tokens[2]; storageIndex = 2; }
+            else { tokenToUse = s_tokens[0]; storageIndex = 0; }
 
             return tokenToUse;
         }
 
         /// <summary>
-        /// This will return up to 3 tokens for the storage accounts
+        /// This will return up to 3 s_tokens for the storage accounts
         /// </summary>
         /// <returns></returns>
-        private string[] GetContainerSasUris()
+        private void GetContainerSasUris()
         {
-            // We need to refresh the tokens every time or they will become invalid.
-            tokens = new string[3];
-            containerAddresses = new string[3];
+            // We need to refresh the s_tokens every time or they will become invalid.
+            s_tokens = new string[3];
+            s_containerAddresses = new string[3];
 
             string accountName = _configuration.GetSection("StorageAccountName")?.Value;
             string accountKey = _configuration.GetSection("StorageAccountKey")?.Value;
-
-            SharedAccessBlobPolicy adHocPolicy = new SharedAccessBlobPolicy()
-            {
-                SharedAccessExpiryTime = DateTime.UtcNow.AddHours(24),
-                Permissions = SharedAccessBlobPermissions.Read
-            };
-
-            containerAddresses[0] = _configuration.GetSection("StorageContainerAddress")?.Value.ToLower();
-            CloudBlobContainer container = new CloudBlobContainer(new Uri(containerAddresses[0]), new StorageCredentials(accountName, accountKey));
-            tokens[0] = container.GetSharedAccessSignature(adHocPolicy, null);
-
-            // Get token for second indexer data source
-            containerAddresses[1] = _configuration.GetSection("StorageContainerAddress2")?.Value.ToLower();
-            if (!String.Equals(containerAddresses[1], defaultContainerUriValue))
+            StorageSharedKeyCredential storageSharedKeyCredential = new StorageSharedKeyCredential(accountName, accountKey);
+            s_containerAddresses[0] = _configuration.GetSection("StorageContainerAddress")?.Value.ToLower();
+            s_containerAddresses[1] = _configuration.GetSection("StorageContainerAddress2")?.Value.ToLower();
+            s_containerAddresses[2] = _configuration.GetSection("StorageContainerAddress3")?.Value.ToLower();
+            int s_containerAddressesLength = s_containerAddresses.Length;
+            if (!String.Equals(s_containerAddresses[1], defaultContainerUriValue))
```

Comment:
I believe based on your current logic this should be:

```csharp
if (String.Equals(s_containerAddresses[1], defaultContainerUriValue))
```
rather than
```csharp
if (!String.Equals(s_containerAddresses[1], defaultContainerUriValue))
```
This function should work whether just one or all three storage accounts all used.